Updated docs (why Chart.js + getting started + step-by-step guide) (#10816)
chartjs/Chart.js: docs/index.md

# Chart.js

Welcome to Chart.js!

* **[Get started with Chart.js](./getting-started/) — best if you're new to Chart.js**
* Migrate from [Chart.js v3](./migration/v4-migration.html) or [Chart.js v2](./migration/v3-migration.html)

> * Join the community on [Slack](https://chartjs-slack.herokuapp.com/) and [Twitter](https://twitter.com/chartjs)

Reviewer: I would keep the Twitter handle away, did not know there was one and it seems to be inactive.

Rather have people to go to slack which in my opinion is clearer and easter for multiple people to react

Reviewer 2: @etimberg has access to that Twitter account, right? I think we have promote releases there, so I'd be fine having it here.

Author: @LeeLenaleee Great that you've spotted this! I wanted to start a conversation about the Twitter account 😄 

Indeed, there hasn't been a lot of activity on https://twitter.com/chartjs in the last ~18 months. At the same time, Slack and Twitter support quite different scenarios: the former is more a community comms platform while the latter is more a news channel.

I believe that we can try reviving Chart.js account on Twitter and maybe grow it to the audience size of https://twitter.com/d3js_org 🤷

I can volunteer to take care of it. @etimberg, WDYT?

As the very first step, I would suggest:
* posting major/minor release updates
* posting highlights about important and/or actively discussed PRs/issues
* posting links to interesting GitHub Discussions/StackOverflow questions
* re-posting other tweets mentioning Chart.js

Reviewer 3: Yes, I have the twitter login. Happy to share access since I don't do much with it. We get tagged by someone else every few days, but it's mostly folks showing off projects that they've built

> * Join the community on [Slack](https://chartjs-slack.herokuapp.com/) and [Twitter](https://twitter.com/chartjs)

Reviewer 3: re slack, I wonder if we should consider migrating from Slack to Discord

* [Contribute to Chart.js](./developers/contributing.html)

## Why Chart.js

Among [many charting libraries](https://awesome.cube.dev/?tools=charts&ref=eco-chartjs) for JavaScript application developers, Chart.js is currently the most popular one according to [GitHub stars](https://github.com/chartjs/Chart.js) (~60,000) and [npm downloads](https://www.npmjs.com/package/chart.js) (~2,400,000 weekly).

Chart.js was created and [announced](https://twitter.com/_nnnick/status/313599208387137536) in 2013 but has come a long way since then. It’s open-source, licensed under the very permissive [MIT license](https://github.com/chartjs/Chart.js/blob/master/LICENSE.md), and maintained by an active community.

### Features

Chart.js provides a set of frequently used chart types, plugins, and customization options. In addition to a reasonable set of [built-in chart types](./charts/area.html), you can use additional community-maintained [chart types](https://github.com/chartjs/awesome#charts). On top of that, it’s possible to combine several chart types into a [mixed chart](./charts/mixed.html) (essentially, blending multiple chart types into one on the same canvas).

> There’s also a [plugin directory](https://github.com/chartjs/awesome#plugins) where you can find add-ons for annotations, zoom, or drag-and-drop.

Reviewer: Something like this sounds a bit better to me, especially the `plugin directory` feels weird. Feels more like that would refer to this: https://github.com/chartjs/Chart.js/tree/master/src/plugins
```suggestion
Chart.js is highly customizable with [custom plugins](https://github.com/chartjs/awesome#plugins) to create annotations, zoom, or drag-and-drop functionalities to name a few things.
```

### Defaults

> Chart.js comes with a sound default configuration, making it very easy to start with and get an app that is ready for production. Chances are you will get a very appealing chart even if you don’t specify any options at all. For instance, Chart.js has animations toggled on by default, so you can instantly bring attention to the story you’re telling with data.

Reviewer: ```suggestion
Chart.js comes with a sound default configuration, making it very easy to start with and get an app that is ready for production. Chances are you will get a very appealing chart even if you don’t specify any options at all. For instance, Chart.js has animations turned on by default, so you can instantly bring attention to the story you’re telling with the data.
```

### Integrations

> Chart.js comes with built-in TypeScript typings and is compatible with all popular [JavaScript frameworks](https://github.com/chartjs/awesome#javascript) including [React](https://github.com/reactchartjs/react-chartjs-2), [Vue](https://github.com/apertureless/vue-chartjs/), [Svelte](https://github.com/SauravKanchan/svelte-chartjs), and [Angular](https://github.com/valor-software/ng2-charts). You can use Chart.js directly or leverage well-maintained wrapper packages that allow for more native integration with your frameworks of choice.

Reviewer: ```suggestion
Chart.js comes with built-in TypeScript typings and is compatible with all popular [JavaScript frameworks](https://github.com/chartjs/awesome#javascript) including [React](https://github.com/reactchartjs/react-chartjs-2), [Vue](https://github.com/apertureless/vue-chartjs/), [Svelte](https://github.com/SauravKanchan/svelte-chartjs), and [Angular](https://github.com/valor-software/ng2-charts). You can use Chart.js directly or leverage on well-maintained wrapper packages that allow for a more native integration with your frameworks of choice.
```

### Developer experience

Chart.js has very thorough documentation (yes, you're reading it), [API reference](./api/), and [examples](./samples/information.html). Maintainers and community members eagerly engage in conversations on [Slack](https://chartjs-slack.herokuapp.com), [GitHub Discussions](https://github.com/chartjs/Chart.js/discussions), and [Stack Overflow](https://stackoverflow.com/questions/tagged/chart.js) where more than 11,000 questions are tagged with `chart.js`.

### Canvas rendering

> Chart.js renders chart elements on HTML5 canvas unlike several other, mostly D3.js-based, charting libraries that render as SVG. Canvas rendering makes Chart.js very performant, especially for large datasets and complex visualizations that would otherwise require thousands of SVG nodes in the DOM tree. At the same time, canvas rendering disallows CSS styling, so you have to use built-in options for that.

Reviewer: ```suggestion
Chart.js renders chart elements on a HTML5 canvas unlike several other, mostly D3.js-based, charting libraries that render as SVG. Canvas rendering makes Chart.js very performant, especially for large datasets and complex visualizations that would otherwise require thousands of SVG nodes in the DOM tree. At the same time, canvas rendering disallows CSS styling, so you will have to use built-in options for that, or create a custom plugin or chart type to render everything to your liking.
```

### Performance

Chart.js is very well suited for large datasets. Such datasets can be efficiently ingested using the internal format so you can skip data [parsing](./general/performance.html#parsing) and [normalization](./general/performance.html#data-normalization). Alternatively, [data decimation](./configuration/decimation.html) can be configured to sample the dataset and reduce its size before rendering.

> In the end, Chart.js uses canvas for efficient rendering, reducing the toll on your DOM tree in comparison to SVG rendering. Also, tree-shaking support allows you to include minimal parts of Chart.js code into your bundle, reducing bundle size and page load time.

Reviewer: Feels like a repeat of the previous part, maby we can combine them 🤔

Author: I've changed a few words to make it sound more natural.
